### ISSUE TITLE
Large amplification of import causes more disk to be used than needed 

### ISSUE THREAD

**Issue by awoods187** [cockroachdb/cockroach#32088]
**Describe the problem**

Node crashed during tpc-c fixtures import on aws. 
![image](https://user-images.githubusercontent.com/22278911/47825060-564aec00-dd46-11e8-8726-bf76c816d9d0.png)
```
F181031 23:27:07.674136 156 storage/store.go:3738  [n2,s2] while committing batch: while committing batch: IO error: No space left on deviceWhile appending to file: /mnt/data1/cockroach/024178.sst: No space left on device
goroutine 156 [running]:
github.com/cockroachdb/cockroach/pkg/util/log.getStacks(0xc420057200, 0xc4200572c0, 0x4132500, 0x10)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:997 +0xcf
github.com/cockroachdb/cockroach/pkg/util/log.(*loggingT).outputLogEntry(0x4a30780, 0xc400000004, 0x4132563, 0x10, 0xe9a, 0xc4275c2000, 0xaa)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:864 +0x8fe
github.com/cockroachdb/cockroach/pkg/util/log.addStructured(0x31cb5e0, 0xc420c10180, 0x4, 0x2, 0x2c32e5b, 0x6, 0xc430f99e60, 0x2, 0x2)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/structured.go:85 +0x2e5
github.com/cockroachdb/cockroach/pkg/util/log.logDepth(0x31cb5e0, 0xc420c10180, 0x1, 0x4, 0x2c32e5b, 0x6, 0xc430f99e60, 0x2, 0x2)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/log.go:69 +0x8c
github.com/cockroachdb/cockroach/pkg/util/log.Fatalf(0x31cb5e0, 0xc420c10180, 0x2c32e5b, 0x6, 0xc430f99e60, 0x2, 0x2)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/log.go:172 +0x7e
github.com/cockroachdb/cockroach/pkg/storage.(*Store).processReady(0xc420710580, 0x31cb5e0, 0xc420c10180, 0x1618)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/store.go:3738 +0x53a
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).worker(0xc4207bc700, 0x31cb5e0, 0xc420c10180)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:214 +0x259
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).Start.func2(0x31cb5e0, 0xc420c10180)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:165 +0x3e
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc42009e510, 0xc420a921b0, 0xc42009e500)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:199 +0xe9
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:192 +0xad
```

Note that there is plenty of space but at least one of the other nodes is very close to full:

![image](https://user-images.githubusercontent.com/22278911/47825238-2f40ea00-dd47-11e8-987d-ee0dc0e0fa06.png)

**To Reproduce**

`roachprod create andy-tpcc-16 -n 4 --clouds=aws  --aws-machine-type-ssd=m5d.4xlarge`
`roachprod stage andy-tpcc-16:1-3 cockroach`
`roachprod stage andy-tpcc-16:4 workload`
`roachprod start andy-tpcc-16:1-3`
`roachprod sql andy-tpcc-16:1`
`SET CLUSTER SETTING kv.range_merge.queue_enabled = false;`
`roachprod run andy-tpcc-16:1 -- "./cockroach workload fixtures import tpcc --warehouses=2500 --db=tpcc"`
[Logs.zip](https://github.com/cockroachdb/cockroach/files/2536739/Logs.zip)

**Environment:**
 - CockroachDB master 10/31/18

**Comment by awoods187** [comment]
Right but that shouldn't have happened as there is plenty of space on the cluster overall. We pointed too much of the import/restore at one disk. TPC-C 2500 is 200gb unreplicated and ~700gb replicated. The m5d.4xlarge has 260gb disk space so * 3 is 780gb.

**Comment by nvanbenschoten** [comment]
This is a three node cluster though, so one node was just lagging behind - there wasn't an imbalance in the restore distribution. The third node would have needed the same capacity as the others eventually, and it looks like they each don't have enough.

**Comment by awoods187** [comment]
One of the nodes only has 160gb in the above screenshot compared to the other at 243gb. Obviously, as we get close to the limits things get dicier but that's 80gbs of difference in two of the nodes. And we should be able to support this number (and did on an earlier run). 

How is that not an imbalance? How are imbalance and lagging behind different?

**Comment by nvanbenschoten** [comment]
I think you're missing my point. We replicate data 3 ways, so the node with less disk usage at the moment can't relieve the node that's further along with the import. It will just eventually get a copy of the same data.

**Comment by awoods187** [comment]
So we spoke offline and I was a bit misunderstanding how the restore works in terms of presenting data in use and the replicated/unreplicated data differences.

While we expect the unreplicated data to be 200gb on one node once fully imported (and for each node to be able to handle up to 260gb), due to compression, write amplification, extra disk usage for other things, it can much more than the expected unreplicated data size. Further, we don't do the best job of playing nicely as we approach the top end of disk availability. These factors together caused the node in question to fill up and fail.

Even so, I'm surprised that 30% wasn't enough buffer to complete the restore. There is some variance involved here because other restore/imports have completed on similar clusters.

**Comment by danhhz** [comment]
It looks like we allude to this in the docs, but it could be more explicit: https://www.cockroachlabs.com/docs/stable/import.html#available-storage

**Comment by awoods187** [comment]
I'm restoring 5k warehouses on a six node cluster right now--it will finish at 1 tb...I'm near the end of the restore and just hit 1.5TBs--that's a lot of amplification

**Comment by awoods187** [comment]
I wonder if this amplification (and reducing it down) is the source of https://github.com/cockroachdb/cockroach/issues/23509

**Comment by dt** [comment]
I think this (import using more storage while it was running) was fixed in v19.2 when we switched to "direct" import that didn't need to buffer (to sort) all the import data on disk before handing it to RocksDB.
